src/lib.rs

```diff
@@ -796,6 +796,35 @@ impl Context {
         }
     }
 
+    /// Test if the given parameters are supported by the TPM.
+    ///
+    /// # Errors
+    /// * if any of the public parameters is not compatible with the TPM,
+    /// an `Err` containing the specific unmarshalling error will be returned.
+    pub fn test_parms(&mut self, parms: PublicParmsUnion) -> Result<()> {
+        let public_parms = TPMT_PUBLIC_PARMS {
+            type_: parms.object_type(),
+            parameters: parms.into(),
+        };
+        let ret = unsafe {
+            Esys_TestParms(
+                self.mut_context(),
+                self.sessions.0,
+                self.sessions.1,
+                self.sessions.2,
+                &public_parms,
+            )
+        };
+
+        let ret = Error::from_tss_rc(ret);
+        if ret.is_success() {
+            Ok(())
+        } else {
```

Comment:
Do we care about differenciating between the parameters not being compatible and a true error in the TSS stack? If so maybe a `Result<bool>` and a different method name (`are_parms_supported`) would be better but if not nevermind.

Reply:
From the specs (the "Commands + Code" document):

> This command is used to check to see if specific combinations of algorithm parameters are supported.The  TPM  will  unmarshal  the  provided  TPMT_PUBLIC_PARMS.  If  the  parameters  unmarshal  correctly, then the TPM will return TPM_RC_SUCCESS, indicating that the parameters are valid for the TPM. The TPM will return the appropriate unmarshaling error if a parameter is not valid.

It might be too complicated for now to make the difference between the two. I thought a specific error would be returned if the parameters were not supported. Ignore this comment!

Reply:
Yeah, I was gonna look up if there was any specific response code that matches that description (although, even then, there's no guarantee that the unmarshalling was wrong because the params are not supported, or because there's something wrong with how the structure was assembled.....)